stackrox/stackrox
# ROX-22405: Replace init bundle wizard with form and modal

## Description

### Problems

* Testers David and Linda found it frustrating to see overview of instructions in step 2 and then not after clicking **Download**.
* It is necessary for **Download** to confirm creating an init bundle.
* Redirect in `MainPage` element to /main/clusters route makes it hard to open other UI pages in a new browser tab.

### Analysis

**Create bundle** is doing too much and current interaction displays instructions too soon.

### Solution

Mansur provided a brilliant simplification:
1. Replace wizard with form which consists of former step 1 and **Download** button.
2. Replace former step 2 of wizard with modal with tabs analogous to SecureClusterPage.
3. Replace links on NoClustersPage with **Review installation methods** button to open modal.

### Residue

1. Verify correctness of content in `SecureClusterUsingOperator` and `SecureClusterUsingHelmChart` elements.
2. Brainstorm how to make the transition from no clusters to clusters clearer when modal is open.

## Checklist
- [x] Edited regression test

## Testing Performed

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.css: 70 = 3947353 - 3947283
        total: 249 = 10516550 - 10516301
    * `ls -al build/static/js/*.js | wc -l`
        0 = 92 - 92 files
3. `yarn start` in ui

### Manual testing

Temporarily edit code to simulate no clusters and no bundles.

1. Visit /main/clusters with simulated no clusters and no bundles.
    * See **Create bundle** button and link.
    ![NoClustersPage_without_bundles](https://github.com/stackrox/stackrox/assets/11862657/03e62667-3d8e-4ca3-80fb-f85c586dd1ba)

2. Click **Legacy installation method** link.
    * See /main/clusters/new page and then click the browser back button.
    ![Legacy_installation_method](https://github.com/stackrox/stackrox/assets/11862657/30af9a1e-6e9f-4d82-b8ac-c86bb981444b)

3. Click **Create bundle** button.
    * See /main/clusters/init-bundles?action=create form with **Download** button disabled.
    ![InitBundleForm_Download_disabled](https://github.com/stackrox/stackrox/assets/11862657/39931755-5beb-4de9-926b-bad4cabf258a)

4. Type name.
    * See **Download** button enabled.
    ![InitBundleForm_Download_enabled](https://github.com/stackrox/stackrox/assets/11862657/62e7fbe5-9d4f-48af-98c1-493606008ff6)

5. Click **Download** button.
    * See **Review installation methods** button instead of **Create bundle** button.
![NoClustersForm_with_bundles](https://github.com/stackrox/stackrox/assets/11862657/5aa015f5-0eaa-48ba-917b-0d419a1d07b3)

6. Click **Review installation methods** button.
    * See modal with **Operator** tab selected.
    ![Modal_Operator](https://github.com/stackrox/stackrox/assets/11862657/11e84d5d-b2bc-4a14-8b62-5d578d5b45ae)

7. Click **Helm chart** tab.
    ![Modal_Helm](https://github.com/stackrox/stackrox/assets/11862657/917317a7-97bd-4501-b5ae-d387773c578f)

8. Click **Operator** tab and remove simulation of no clusters.
    * See clusters page under modal.
    ![Modal_ClustersTablePanel](https://github.com/stackrox/stackrox/assets/11862657/1aa1e083-5c35-4fc6-99a9-dbbc42a9e37a)

9. Click **Close** button.

10. Click **Secure a cluster** button at upper right of clusters page.
    ![Secure_a_cluster_button](https://github.com/stackrox/stackrox/assets/11862657/31bd532d-99bc-45b5-bd01-cb2668613405)

11. Click **Init bundle installation methods** item.
    * See tabs in /main/clusters/secure-a-cluster page similar to modal.
    ![Secure_a_cluster_ppage](https://github.com/stackrox/stackrox/assets/11862657/7e21dc28-0729-4bb7-b53f-67ca61d93859)

### Integration testing

Before you `yarn cypress-open` in ui/apps/platform folder:

```sh
export CYPRESS_ROX_MOVE_INIT_BUNDLES_UI=true
```

* clusters/redirectFromDashboard.test.js
